Goal: fix memory leak in vfs_full_audit

Fixes: #520794

Status wrt upstream: Fixed in 3.2.8

Author: Volker Lendecke <vl@samba.org>

Index: lenny/source/modules/vfs_full_audit.c
===================================================================
--- lenny.orig/source/modules/vfs_full_audit.c
+++ lenny/source/modules/vfs_full_audit.c
@@ -701,6 +701,7 @@
 static char *audit_prefix(TALLOC_CTX *ctx, connection_struct *conn)
 {
 	char *prefix = NULL;
+	char *result;
 
 	prefix = talloc_strdup(ctx,
 			lp_parm_const_string(SNUM(conn), "full_audit",
@@ -708,12 +709,14 @@
 	if (!prefix) {
 		return NULL;
 	}
-	return talloc_sub_advanced(ctx,
+	result = talloc_sub_advanced(ctx,
 			lp_servicename(SNUM(conn)), conn->user,
 			conn->connectpath, conn->gid,
 			get_current_username(),
 			current_user_info.domain,
 			prefix);
+	TALLOC_FREE(prefix);
+	return result;
 }
 
 static bool log_success(vfs_handle_struct *handle, vfs_op_type op)
